lib/executor/ramping_arrival_rate.go

```diff
@@ -175,11 +175,23 @@ func (varc RampingArrivalRateConfig) HasWork(et *lib.ExecutionTuple) bool {
 type RampingArrivalRate struct {
 	*BaseExecutor
 	config RampingArrivalRateConfig
+	et     *lib.ExecutionTuple
 }
 
 // Make sure we implement the lib.Executor interface.
 var _ lib.Executor = &RampingArrivalRate{}
 
+// Init values needed for the execution
+func (varr *RampingArrivalRate) Init(ctx context.Context) error {
+	// err should always be nil, because Init() won't be called for executors
+	// with no work, as determined by their config's HasWork() method.
+	et, err := varr.BaseExecutor.executionState.ExecutionTuple.GetNewExecutionTupleFromValue(varr.config.MaxVUs.Int64)
+	varr.et = et
+	varr.iterSegIndex = lib.NewSegmentedIndex(et)
+
+	return err //nolint: wrapcheck
+}
```

Comment:
Hmm this `Init()` makes sense, I think, but I wonder why we didn't have this re-segmentation before? Was that a bug we have to mention in the release notes separately?

Reply:
Not sure, it seems to segment properly to me on `master`, but maybe I'm not reproducing it.

If it was a minor bug we should mention it, sure.

Reply:
Apparently, I didn't do part of https://github.com/k6io/k6/issues/1308 :sweat: for this executor.

With 
```javascript
export let options = {
        scenarios: {
                contacts: {
                        executor: 'ramping-arrival-rate',
                        startRate: 5,
                        timeUnit: '1s',
                        preAllocatedVUs: 2,
                        maxVUs: 3,
                        stages:[ { target: 5, duration: "3s"}],
                },
        },
};
export default function () {
    console.log(__VU, __ITER);
}
```
(Notice the `maxVUs: 3` and preallocatedVUs: '2', it important and they both can't be proportionally split given a segment)
You can clearly see that with master and `--execution-segment '0:1/2'` it does  7 iterations (there re in total 15 = 3s * 5 iter/s).
With the changes from this branch it does 10 or 2/3 of the iterations as when you scale based on the maxVUs it gets 2 maxVUs out of 3. 
So ... good thing we caught and we should mention it. I doubt anyone actually got bitten by this or even if all that many people experienced it unknowingly, but it's definitely still a bug

Reply:
@imiric can you please add a link to or a copy of :arrow_up: in the PR description, so when we link it in the v0.33.0 release notes, it's clear that it also fixes it?